bazel/envoy_build_system.bzl

```diff
@@ -195,6 +196,40 @@ def envoy_include_prefix(path):
         return "/".join(path.split("/")[1:])
     return None
 
+# External CMake C++ library targets should be specified with this function. This defaults
+# to building the dependencies with ninja
+def envoy_cmake_external(
+        name,
+        cache_entries = {},
+        cmake_options = ["-GNinja"],
+        make_commands = ["ninja", "ninja install"],
+        lib_source = "",
+        postfix_script = "",
+        static_libraries = []):
+    # On Windows, we don't want to explicitly set CMAKE_BUILD_TYPE,
+    # rules_foreign_cc will figure it out for us
+    cache_entries_no_build_type = {}
+    for key in cache_entries.keys():
+        if key != "CMAKE_BUILD_TYPE":
```

Comment:
CMAKE_BUILD_TYPE is calculated by cmake_external internally, from the type of Bazel build.
I added the way how to avoid that: https://github.com/bazelbuild/rules_foreign_cc/pull/224

Please pass "CMAKE_BUILD_TYPE": "" (empty string) to cmake_external, and then the computed value for CMAKE_BUILD_TYPE will not be used (there would be no "-DCMAKE_BUILD_TYPE=<something>" on the cmake call.)

Reply:
On Windows, we want to use the computed value for CMAKE_BUILD_TYPE so we can get the desired release / debug builds